### DISCUSSION TITLE
Connecting to snowflake through benthos

### DISCUSSION THREAD

**Discussion by seegop** [redpanda-data/connect#2569] [Q&A]
input:

  sql_select:

    driver: snowflake

    dsn: username:pwd@account/dbname/warehousename?&authenticator=snowflake_jwt&privateKey=$$$

    table: "tablename"

    columns:

             - tablename.columnname

    output:
         stdout:
             codec:lines

I get a timeout error
failed to connect to sql_select: timeout after 5m0s and 8 retries..any help??

 

 

 

output:

    stdout:

        codec: lines

**Comment by mihaitodor** [comment]
Hey @seegop 👋 Hard to say what's wrong... My best guess is that you're setting up the DSN wrong. Please consult the [gosnowflake docs](https://pkg.go.dev/github.com/snowflakedb/gosnowflake#hdr-Connection_String) for more details. Note that some of the parameters in the DSN need to be escaped as described in the Benthos docs [here](https://www.benthos.dev/docs/components/inputs/sql_select/#drivers) (see the Snowflake note under the drivers table).
